include/zephyr/toolchain/gcc.h

```diff
@@ -14,6 +14,15 @@
  * Macros to abstract compiler capabilities for GCC toolchain.
  */
 
+/*
+ * gcc.h is included indirectly from devicetree.h and uses autoconf.h.
```

Comment:
This is confusing and not clear, not sure how devicetree.h indirectly include gcc.h requires us to add an explicti include to a file that is not being included anywhere and is handled by the build system. We never needed to include autoconf.h directly anywhere...

Reply:
devicetree.h is used by non-zephyr build systems such as bootloaders and secure root-of-trust images..

These build systems may not be doing "-include autoconf.h".

Adding an #include for autoconf.h is a harmless addition to gcc.h that improves integration between Zephyr and non-zephyr build systems.

Reply:
If they make use of Zephyr headers out of Zephyr, they must take care of any dependency, be `-imacros`, other include paths, etc.